src/fswAlgorithms/orbitControl/lambertSolver/lambertSolver.cpp

```diff
@@ -242,6 +242,43 @@ std::vector<double> LambertSolver::householder(double T, double x0, int N)
     return sol;
 }
 
+/*! This method includes a halley root-finder (2nd order householder) to find the free variable x that satisfies the time-of-flight constraint
+    @param T requested non-dimensional time-of-flight
+    @param x0 initial guess for x free variable of Lambert's problem that satisfies the given time of flight
+    @param N number of revolutions
+    @return std::vector<double>
+*/
+std::vector<double> LambertSolver::halley(double T, double x0, int N)
+{
+    double tol = 1e-8;
+    int iter_max = 8;
+    double xnew = 0.0; // initialize
+    int j;
+    double err;
+
+    for (j = 0; j <= iter_max; ++j){
```

Comment:
:large_orange_diamond: Declare the incremented variable inside the for loop statement.